### PR TITLE
Some small improvements to the Mushrooms example

### PR DESCRIPTION
**Pull request by tgpfeiffer**
I found the introductory sentence to the code snippet that introduces the `trainer` quite confusing, so I rewrite it. Also I added one introductory sentence to the `Extensions` section. Please check if the explanation is appropriate and understandable, thanks.

### CODE REVIEW

**Comment by hvy** [comment]
@Crissman could you please take a look?

File: docs/source/glance.rst [hunk 1]
```diff
@@ -135,15 +135,15 @@ Updater
...
 
 .. literalinclude:: ../../examples/glance/glance.py
    :language: python
    :linenos:
    :lines: 49-50
    :lineno-start: 49
 
-Set up the :class:`updater <chainer.training.Updater>` to be called after the training batches and set the number of batches per epoch to 100. The learning rate per epoch will be output to the directory ``result``.
+Finally we create a :class:`Trainer <chainer.training.Trainer>` object. The ``trainer`` processes minibatches using the ``updater`` defined above until a certain stop condition is met, and allows to use extensions at certain phases of the training. We set it up to run for 50 epochs and store all files created by extensions (see below) in the ``result`` directory.
```
Comment (Crissman) [inline docs/source/glance.rst]:
```suggestion
Finally we create a :class:`Trainer <chainer.training.Trainer>` object. The ``trainer`` processes minibatches using the ``updater`` defined above until a certain stop condition is met and allows the use of extensions during the training. We set it to run for 50 epochs and store all files created by the extensions (see below) in the ``result`` directory.
```

File: docs/source/glance.rst [hunk 2]
```diff
@@ -158,9 +158,9 @@ Extensions
    :width: 835px
    :height: 353px
 
-Use the testing :class:`iterator <chainer.dataset.Iterator>` defined above for an :class:`~chainer.training.extensions.Evaluator` extension to the trainer to provide test scores.
+Extensions can be used to execute code at certain events during the training, for example every epoch or every 1000 iterations. This mechanism is used in Chainer to evaluate models during training, print progress messages, or dump intermediate model files.
```
Comment (Crissman) [inline docs/source/glance.rst]:
```suggestion
Extensions can be used to execute code at certain events during the training, such as every epoch or every 1000 iterations. This mechanism is used in Chainer to evaluate models during training, print progress messages, or dump intermediate model files.
```

**Comment by tgpfeiffer** [comment]
@Crissman Thanks for your review! I incorporated your suggestions.

**Comment by Crissman** [comment]
LGTM. Merging!

**Pull request merged**